server/models/contests/getContestDetails.js

```diff
@@ -25,7 +25,20 @@ function getContestDetails({ username, params, query }) {
           'Invalid contest Id or the user do not have required permissions'
         )
       }
-      return resolve(results)
+      if (onlyName) {
+        return resolve(results[0])
+      }
+      pool.query(
+        `SELECT group_id, group_name, member_count FROM \`groups\` g INNER JOIN contests_groups cg
+      ON g.id=cg.group_id WHERE cg.contest_id=?`,
```

Comment:
Here, transaction should not be required, as both the queries we are executing are **read queries** and are **independent**. Here, it is not the situation like, if one query fails, then the other query should be rolled back, because both are **read queries**. Starting a transaction locks all other connections from being able to execute any SQL until the transaction is committed or rolled back. So, unnecessary transactions may slow down other queries.

I found [this](https://stackoverflow.com/questions/308905/should-there-be-a-transaction-for-read-queries) for more details on when we should use transaction for read queries.